klpbuild/plugins/extractor.py

```diff
@@ -341,23 +341,25 @@ def apply_all_patches(self, cs, fil):
         if not patched:
             raise RuntimeError(f"{cs.name()}({cs.kernel}): Failed to apply patches. Aborting")
 
-
-
     def get_cmd_from_json(self, cs, fname):
-        cc_file = Path(cs.get_obj_dir(), "compile_commands.json")
-        # FIXME: compile_commands.json that is packaged with SLE/openSUSE
-        # doesn't quite work yet, so don't use it yet.
-        return None
+        cc_file = cs.get_obj_dir()/"compile_commands.json"
+
+        # Older codestreams doens't support compile_commands.json, so use make for them
+        if not cc_file.exists():
+            return None
 
         with open(cc_file) as f:
             buf = f.read()
         data = json.loads(buf)
         for d in data:
             if fname in d["file"]:
                 output = d["command"]
-                return Extractor.process_make_output(output)
+                # The arguments found on the file point to .., since they are generated
+                # when the kernel is compiled. Replace the .. by the codestream kernel source
+                # directory since klp-ccp needs to reach the files
+                return Extractor.process_make_output(output).replace("..", str(cs.get_src_dir()))
 
-        logging.error(f"Couldn't find cmdline for {fname}. Aborting")
+        logging.error("Couldn't find cmdline for %s. Aborting", fname)
```

Comment:
Perhaps I'd rather not use here "Aborting", since if we don't find the command-line we're looking for in the json we just use the old method, right?
We could just emit a warning instead of an error, like:
```suggestion
        logging.warning("Couldn't find cmdline for %s in compile_commands.json, using get_make_cmd().", fname)
```
or something similar.

Reply:
We can do that, but we won't expect the make approach to work either, since compile_commands.json is created for each compiled source file. We can do what you suggested, but I won't expect to have more value from it.

Reply:
I agree with you on this, my concern is just that as far as I understand we're not aborting here. Unless I'm missing something